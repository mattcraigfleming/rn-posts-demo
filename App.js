import React from 'react';
import {SafeAreaView, StyleSheet, View, Text} from 'react-native';

const App = () => {
  return (
    // SafeAreaView for iOS version 11 or later https://facebook.github.io/react-native/docs/safeareaview
    <SafeAreaView>
      <View style={styles.body}>
        <Text>Home</Text>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: 'white',
  },
});

export default App;
