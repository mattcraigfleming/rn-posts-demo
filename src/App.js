import React from 'react';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/integration/react';

import {store, persistor} from './store';
import Navigator from './navigation/Navigator';

const App = () => {
  return (
    <Provider store={store}>
      <Navigator />
    </Provider>
  );
};

export default App;
