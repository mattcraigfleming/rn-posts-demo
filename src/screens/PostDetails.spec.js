import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';

jest.mock('react-native/Libraries/Lists/FlatList', () => 'FlatList');

// imported as a connected component!
import App from '../App';

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {
  posts: [],
};

describe('Testing Authors', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<App />, {
      context: {store: mockStore(initialState)},
    }).dive();
    expect(wrapper.dive()).toMatchSnapshot();
  });
});

//   const leanne = {id: 1, name: 'Leanne Graham'};
//   const ervin = {id: 2, name: 'Ervin Howell'};
//   const authors = [leanne, ervin];
