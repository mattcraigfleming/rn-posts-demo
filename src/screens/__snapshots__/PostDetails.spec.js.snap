// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authors renders correctly 1`] = `null`;

exports[`Testing Authors renders as expected 1`] = `ShallowWrapper {}`;
